pkg/backup/backupdest/incrementals_test.go

```diff
@@ -117,3 +125,131 @@ func TestJoinURLPath(t *testing.T) {
+		name          string
+		paths         []backupPath
+		expectedPaths []string
+	}
+
+	for idx, tc := range []testcase{
+		{
+			name: "all suffixed paths",
+			paths: []backupPath{
+				{path: "/20250320/001000.00-20250320-000000.00", useOldBackup: false},
+				{path: "/20250320/002000.00-20250320-001000.00", useOldBackup: false},
+				{path: "/20250320/003000.00-20250320-002000.00", useOldBackup: false},
+			},
+			expectedPaths: []string{
+				"/20250320/001000.00-20250320-000000.00",
+				"/20250320/002000.00-20250320-001000.00",
+				"/20250320/003000.00-20250320-002000.00",
+			},
+		},
+		{
+			name: "all non-suffixed paths",
+			paths: []backupPath{
+				{path: "/20250320/001000.00", useOldBackup: false},
+				{path: "/20250320/002000.00", useOldBackup: false},
+				{path: "/20250320/003000.00", useOldBackup: false},
+			},
+			expectedPaths: []string{
+				"/20250320/001000.00",
+				"/20250320/002000.00",
+				"/20250320/003000.00",
+			},
+		},
+		{
+			name: "all old backup paths",
+			paths: []backupPath{
+				{path: "/20250320/001000.00", useOldBackup: true},
+				{path: "/20250320/002000.00", useOldBackup: true},
+				{path: "/20250320/003000.00", useOldBackup: true},
+			},
+			expectedPaths: []string{
+				"/20250320/001000.00",
+				"/20250320/002000.00",
+				"/20250320/003000.00",
+			},
+		},
+		{
+			name: "mixed new and old backup paths",
+			paths: []backupPath{
+				{path: "/20250320/001000.00", useOldBackup: false},
+				{path: "/20250320/002000.00", useOldBackup: true},
+				{path: "/20250320/003000.00", useOldBackup: false},
+			},
+			expectedPaths: []string{
+				"/20250320/001000.00",
+				"/20250320/002000.00",
+				"/20250320/003000.00",
+			},
+		},
+		{
+			name: "mixed suffixed and non-suffixed paths",
+			paths: []backupPath{
+				{path: "/20250320/001000.00-20250320-000000.00", useOldBackup: false},
+				{path: "/20250320/002000.00", useOldBackup: false},
+				{path: "/20250320/003000.00-20250320-002000.00", useOldBackup: false},
+			},
+			expectedPaths: []string{
+				"/20250320/001000.00-20250320-000000.00",
+				"/20250320/002000.00",
+				"/20250320/003000.00-20250320-002000.00",
+			},
+		},
+		{
+			name: "invalid backup paths with valid backup paths",
+			paths: []backupPath{
+				{path: "/20250320/001000.00-20250320-000000.00", useOldBackup: false},
+				{path: "/20250320/001500.000", useOldBackup: false}, // invalid
+				{path: "/20250320/002000.00", useOldBackup: false},
+				{path: "/2025/03/20/002500.00", useOldBackup: false}, // invalid
+				{path: "/20250320/003000.00", useOldBackup: true},
+			},
+			expectedPaths: []string{
+				"/20250320/001000.00-20250320-000000.00",
+				"/20250320/002000.00",
+				"/20250320/003000.00",
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			store, err := execCfg.DistSQLSrv.ExternalStorageFromURI(
+				ctx, fmt.Sprintf("nodelocal://1/%d", idx), username.RootUserName(),
+			)
+			require.NoError(t, err)
+			defer store.Close()
```

Comment:
nit: randomize the order of tc.paths? This will ensure FindPrioriBackups returns stuff in order, regardless of the input.